### PR TITLE
LOG-4676: Modify receiverport to be a simple int instead of a struct

### PR DESCRIPTION
**Pull request by syedriko**
### Description
ReceiverPort is defined as a complex struct which mimics the need of defining a port for a generic service. The CLO controls all aspects of the receiver so it should need to have this complexity; users should only need to define potentially the port upon which to listen. Modify the API and code to convert 'receiverPort' into an int

/cc @alanconway 
/assign @jcantrill 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-4676

### CODE REVIEW

File: internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go [hunk 1]
```diff
@@ -201,10 +201,8 @@ func verifyInputs(spec *loggingv1.ClusterLogForwarderSpec, status *loggingv1.Clu
 			badInput("ReceiverSpec must define an HTTP receiver")
 		case isHTTPReceiver(input) && input.Receiver.HTTP.Format != loggingv1.FormatKubeAPIAudit:
 			badInput("invalid format specified for HTTP receiver")
-		case isHTTPReceiver(input) && isInvalidPort(input.Receiver.HTTP.ReceiverPort.Port):
+		case isHTTPReceiver(input) && isInvalidPort(input.Receiver.HTTP.Port):
```
Comment (jcantrill) [inline internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go:204]: is it going to be a problem to allow 1-1024?  I seem to recall those being reserved ports

**Review by jcantrill — COMMENTED** [review]
looks good except one question.  also please run "make docs"

File: internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go (hunk 1 above)
Comment (syedriko) [reply internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go]: Ran `make docs`, excluded ports [1,1023]

**Pull request merged**